### PR TITLE
feat(components): [date-picker] add icon slots

### PR DESCRIPTION
**Pull request by zhixiaotong**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

343058749-2f48b9a1-3231-463e-9b70-0f49d39fd607

https://github.com/element-plus/element-plus/assets/34681550/a26b1a30-837c-40c6-9386-01a1c79ec077

### CODE REVIEW

**Review by btea — APPROVED** [review]

**Comment by zhixiaotong** [comment]
@btea I changed version ^2.8.0 to ^2.7.8.

**Comment by btea** [comment]
The next version is 2.8.0.

**Comment by zhixiaotong** [comment]
> The next version is 2.8.0.

🙈我待会又改回去

**Comment by zhixiaotong** [comment]
> > The next version is 2.8.0.
> 
> 🙈我待会又改回去

revert complete.

File: docs/en-US/component/datetime-picker.md [hunk 1]
```diff
@@ -73,6 +73,16 @@ datetime-picker/default-time
 
 :::
 
+## Custom icon
```
Comment (tolking) [inline docs/en-US/component/datetime-picker.md:76]:
```suggestion
## Custom icon ^(2.8.0)
```

**Review by tolking — COMMENTED** [review]
After merging #14804, you may also need to deal with the newly added files

**Comment by tolking** [comment]
Check panel-year-range.vue

**Comment by zhixiaotong** [comment]
> Check panel-year-range.vue

okay

**Comment by zhixiaotong** [comment]
https://github.com/user-attachments/assets/3284a4df-4f18-4250-abbc-cee4c3bc740f

done @tolking

File: docs/en-US/component/datetime-picker.md (hunk 1 above)
Comment (tolking) [reply docs/en-US/component/datetime-picker.md:76]: check

**Review by tolking — APPROVED** [review]

**Pull request merged**